### PR TITLE
Add new vulnerabilities resource for IoT Security

### PR DESCRIPTION
**Pull request by galbru**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet: **Late August 2021**
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet: **August 2021**
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json
Comment (liranc) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json:13]: `locations` instead of `location`

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json
Comment (liranc) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json:15]: `locations`

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json
Comment (liranc) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json:17]: What's the purpose of the `id` property? (isn't it the same as the top-level `name` property?)

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json
Comment (liranc) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json:18]: Is there a situation where this title doesn't match the resource name?

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (liranc) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:219]: Some properties are marked as `readOnly` while others don't ... is it on purpose? (`severity`, `score`)

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (liranc) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:231]: Is there an enum that can represent this type?

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (galbru) [reply specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:219]: no. fixed.

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json
Comment (galbru) [reply specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json:18]: I set the resource name with the id of the CVE.

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json
Comment (galbru) [reply specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/examples/Vulnerabilities/GetDeviceVulnerability.json:17]: same

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (orkayam) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:187]: redundant, it's part of the resource

**Comment by j5lim** [comment]
> NewApiVersionRequired reason:
> 
> A service’s API is a contract with customers and is represented by using the api-version query parameter. Changes such as adding an optional property to a request/response or introducing a new operation is a change to the service’s contract and therefore requires a new api-version value. This is critically important for documentation, client libraries, and customer support.
> 
> EXAMPLE: if a customer calls a service in the public cloud using api-version=2020-07-27, the new property or operation may exist but if they call the service in a government cloud, air-gapped cloud, or Azure Stack Hub cloud using the same api-version, the property or operation may not exist. Because there is no clear relationship between the service api-version and the new property/operation, customers can’t trust the documentation and Azure customer have difficulty helping customers diagnose issues. In addition, each client library version documents the service version it supports. When an optional property or new operation is added to a service and its Swagger, new client libraries must be produced to expose this functionality to customers. Without updating the api-version, it is unclear to customers which version of a client library supports these new features.

@galbru Please follow the process in Breaking Change Review Checklist to get an approval from the breaking change review board if you want to keep the api-version.

**Comment by orkayam** [comment]
Hi @Jin Soon ***@***.***>,

This API version is in private preview and was release only for a few customers so far only in public cloud.
Adding a new API version before release to public preview is quite redundant...?

Thanks,
Orel.

**Comment by galbru** [comment]
@jianyexi Breaking change approved. Can you please approve the PR? Thanks.

**Comment by jianyexi** [comment]
> @jianyexi Jianye Xi FTE Breaking change approved. Can you please approve the PR? Thanks.

needs ARM review & sign off

File: specification/iotsecurity/resource-manager/readme.md
Comment (raosuhas) [inline specification/iotsecurity/resource-manager/readme.md:42]: Looks you do support pagination in yout collection call. WHy do you need this ?

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (raosuhas) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:71]: Is it expected for you to support both skip token and the paging header ?

File: specification/iotsecurity/resource-manager/readme.md
Comment (galbru) [reply specification/iotsecurity/resource-manager/readme.md:42]: mistake

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (galbru) [reply specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:71]: I'm adding the paging header key and support skip token.

**Comment by galbru** [comment]
@raosuhas waiting for your feedback

File: specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json
Comment (mentat9) [inline specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-07-01-preview/vulnerabilities.json:67]: >$skipToken [](http://example.com/codeflow?start=21&length=10)

Current naming convention is $skip (this won't block ARM signoff).

**Review by jianyexi — APPROVED** [review]

**Pull request merged**
